Short summary of NuCypher in the README

cygnusv opened the pull request:
This PR is just a basic UX improvement: Don't make people click on a secondary link (e.g., docs, whitepaper, website, etc) in order to learn about what is NuCypher. 

Other stuff in this PR:
- Fixes badges links and unify styles.
- Tries to fix bad render of Markdown text in pypi project description (see #794).

derekpierre commented:
@cygnusv, side note: @arjunhassard was also looking to do something similar regarding the summary in the index.rst file, see https://github.com/nucypher/nucypher/pull/704 although still [WIP].

derekpierre commented on README.md:
```diff
@@ -2,16 +2,36 @@
 ----
 
-![](https://img.shields.io/pypi/v/nucypher.svg?style=flat)
+The NuCypher network uses the [Umbral](https://github.com/nucypher/pyUmbral) threshold proxy re-encryption scheme to provide cryptographic access controls for distributed apps and protocols.
```
Perhaps something after this sentence about the usage by applications? Example:

```suggestion
The NuCypher network uses the [Umbral](https://github.com/nucypher/pyUmbral) threshold proxy re-encryption scheme to provide cryptographic access controls for distributed apps and protocols. Applications can use the NuCypher network to facilitate end-to-end encrypted data sharing via sharing policies. Access permissions are baked into the underlying encryption, and access can only be explicitly granted by the data owner. Consequently, the data owner has ultimate control over access to their data. At no point is the data decrypted nor can the underlying private keys be determined by the NuCypher network.
```

michwill replied:
Good suggestion, indeed

cygnusv commented:
> @cygnusv, side note: @arjunhassard was also looking to do something similar regarding the summary in the index.rst file, see #704 although still [WIP].

Yes, although I think both PRs are complementary. I was just looking for a short summary (~ 100-150 words).

mswilkison commented on README.md:
```diff
@@ -2,16 +2,36 @@
-![](https://img.shields.io/pypi/v/nucypher.svg?style=flat)
+The NuCypher network uses the [Umbral](https://github.com/nucypher/pyUmbral) threshold proxy re-encryption scheme to provide cryptographic access controls for distributed apps and protocols.
+
+01. Alice, the data owner, grants access to her encrypted data to 
```
Wdyt about including diagrams for each step like the ones here: https://github.com/nucypher/slides/blob/master/svg/delegate.svg

tuxxy replied:
Hmm, I think I'd prefer this in the docs, not the README.

cygnusv replied:
I'm with @tuxxy in that a diagram for each step belongs to docs, rather than the README,
but I'm all in for one general diagram for the README.

mswilkison commented on docs/source/index.rst:
```diff
@@ -24,15 +23,37 @@ NuCypher
+If the policy is satisfied, the data is re-encrypted to his public key
+and he can decrypt it with his private key.
+
+More information:
```
```suggestion
More detailed information:
```

cygnusv commented on README.md:
```diff
@@ -2,16 +2,36 @@
+The resulting encrypted data can be uploaded to IPFS, Swarm, S3, 
+or any other storage layer.
+
+03. Ursula, a miner, receives the access policy and stands ready to 
```
What about just `Ursula, a miner in the NuCypher network...`?

derekpierre replied:
My hesitation with just `miner` is for a federated network case given its association with tokens/cryptocurrencies etc. but perhaps it is unwarranted given our current focus.

KPrasch replied:
are PoS nodes properly characterized by "miner"?

cygnusv replied:
We use the miner/mining terminology extensibly in our codebase, so I think at least from our terminology perspective it's justified. In any case, this is copied from the website!

Although, on second thoughts, I think that just `node` might be better:
```
Ursula, a node in the NuCypher network, ...
```

derekpierre replied:
👍

cygnusv commented:
@michwill I don't know about the errors introduced by `py-evm`, but `vodka` pins it to version `0.2.0a39` so perhaps that can solve it. I must say I haven't experienced such errors, nor the CI build.

@derekpierre @mswilkison I addressed most of your suggestions in 6ce6d68 (marked as resolved in this thread), but I'm not so sure about a couple of them. See my comments.

cygnusv commented on README.md:
```diff
@@ -2,16 +2,36 @@
+anyone she wants by creating a policy and uploading it to 
+the NuCypher network.
+
+02. Anyone can encrypt data using Alice's policy public key. 
```
Perhaps we should refine this proposed sentence, since it only gives examples of "small" devices. 
`Examples of entities that can do this are IoT devices, a friend, or Alice herself.`

In this case we're covering both machines and humans. The "a friend" example is weak, but I don't have another idea to convey the meaning of a third-party human entity.

derekpierre replied:
Good point - 👍
I agree "friend" is a little weird, but I can't think of a better word 😅- "associate", "collaborator" (?) - clearly still just synonyms of "friend" but perhaps comes across better...

mswilkison replied:
"comrade"!

arjunhassard replied:
I would rephrase the whole of line 21 as:
`Using her policy's public key, any entity can encrypt data on Alice's behalf. This entity could be an IOT device in her car, a collaborator assigned the task of writing data to her policy, or even a third-party creating data that belongs to her – for example, a lab analyzing medical tests.`

cygnusv replied:
@arjunhassard Sorry, we merged this PR and forgot to address this comment! I'm all in for this propose change. How about you change it in #704?

KPrasch commented on setup.py:
```diff
@@ -112,6 +112,7 @@ def run(self):
       description=ABOUT['__summary__'],
       license=ABOUT['__license__'],
       long_description=long_description,
+      long_description_content_type="text/markdown",
```
:+1:

KPrasch commented on docs/source/index.rst:
```diff
@@ -24,15 +23,37 @@ NuCypher
+the NuCypher network.
+
+2. Anyone can encrypt data using Alice's policy public key.
+The resulting encrypted data can be uploaded to IPFS, Swarm, S3,
```
``` suggestion
The resulting encrypted data can be uploaded to a storage layer like IPFS, Swarm, S3,
```

cygnusv replied:
The next line said: ``or any other storage layer.``
Is that OK?

KPrasch commented:
Rebuilding: https://circleci.com/workflow-run/8906181f-79a8-43ca-8bf5-70b8c06a1070
